react-app/src/components/molecules/MovieDetails/MovieDetailsComponent.tsx

```diff
@@ -20,14 +21,20 @@ import {
 } from './MovieDetails';
 import { actionControlVisibility } from '../../../actions/Actions';
 import { IMovie } from '../MovieCard/IMovie';
+import { IState } from '../../../reducers/IState';
 
 const MovieDetailsComponent: FunctionComponent = () => {
   const type = 'details';
-  let visible = useSelector(({ modals: { [type]: visibility } }) => visibility);
-  const movieDetails: IMovie = useSelector(({ movieDetail: { movieDetail: movie } }) => movie);
+  let visible = useSelector(({ modals: { [type]: visibility } }: IState) => visibility);
```

Comment:
I'd suggest to have a look at reusable selectors with reselect library for performance reasons.

Reply:
Rebuild only one component to use reselect. The rest left with useSelector for example in case I will return to it later.